agent/unit/unit.go

```diff
@@ -0,0 +1,35 @@
+package unit
+
+type Unit struct {
+	Name   string
+	Status Status
+}
+
+type Status string
+
+const (
+	StatusPending   Status = "pending"
+	StatusStarted   Status = "started"
+	StatusCompleted Status = "completed"
+	StatusFailed    Status = "failed"
+)
+
+// DependencyCoordinator is a composition of multiple structures that work together to
+// coordinate the dependencies between units. It is composed of a Graph of Units and Statuses
+// to track propagation of status changes through the graph. It also provides a pub-sub mechanism
+// to notify units when their dependencies are updated such that they can update their own status
+// accordingly.
+type DependencyCoordinator struct {
+	*Graph[Status, *Unit]
+	// TODO(sas): implement pub-sub mechanism to notify units when their dependencies are updated
+}
+
+func (g *DependencyCoordinator) AddEdge(from, to *Unit, edge Status) {
```

Comment:
Suggestion: Consider alternate naming to avoid leaking graph semantics to a higher level. Perhaps something like `AddDependency(dependent, dependency, requiredStatus)` (maybe forward/backward?).

Nits:

- Documentation for method
- Receiver name is not consistent with type